bigchaindb/backend/mongodb/connection.py

```diff
@@ -143,6 +145,11 @@ def _connect(self):
             raise ConfigurationError from exc
 
 
+MONGO_OPTS = {
+    'socketTimeoutMS': 20000,
+}
```

Comment:
Would it help to add `readPreference='nearest'` as the user suggested.
This might let the writes go to the primary and reads to the local MongoDB instance in the replica set.
Helps with decentralization and performance too. Thoughts, @libscott @r-marques ?